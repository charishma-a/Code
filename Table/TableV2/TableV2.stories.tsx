import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import { TableV2 } from './TableV2'
import { device } from '@/utils/device'
import styled from 'styled-components'
import { Tooltip, TooltipIcon, TooltipText } from '@/common/styled'
import { IMAGE_PROPS_INFO_PURPLE } from '@/constants/images'

export default {
  title: 'TableV2',
  component: TableV2,
} as Meta

const defaultProps = {
  columns: [
    {
      accessor: 'category',
      Header: 'Category',
    },
    { accessor: 'description', Header: 'Description' },
    { accessor: 'totalCostUsd', Header: 'Total Cost ($USD)' },
  ],
  data: [
    {
      category: 'Salaries & Benefits',
      description: 'Post-doctoral Fellow',
      totalCostUsd: '$40,000',
    },
    {
      category: 'Materials and Supplies',
      description: (
        <>
          <p style={{ marginTop: 0 }}>
            Flowcytometry, secretome, miRNA seq, mass spectrometry analyses
          </p>
          <p style={{ marginBottom: 0 }}>
            Tissue culture media, Plasticware & cytokines, Antibodies, Cell
            lines
          </p>
        </>
      ),
      totalCostUsd: '$20,000',
    },
    {
      category: 'Travel',
      description: 'Travel to conference for PDF',
      totalCostUsd: '$1,500',
    },
    {
      category: 'Knowledge Translation',
      description: 'Publication costs',
      totalCostUsd: '$3,850',
    },
    {
      category: 'Total Funds',
      description: '',
      totalCostUsd: '$122,350',
      removeEmptyCells: true,
      rowStyles: `
        grid-template-columns:
        minmax(150px, 2fr)
        minmax(150px, 1fr); td { color: #8256FF; }
      `,
    },
    {
      category: 'Research Institute Salaries & Administrative Support – 15%',
      description: '',
      totalCostUsd: '$18,350',
      removeEmptyCells: true,
      rowStyles: `
        grid-template-columns:
        minmax(150px, 2fr)
        minmax(150px, 1fr);
      `,
    },
    {
      category: 'Space – 15%',
      description: '',
      totalCostUsd: '$18,350',
      removeEmptyCells: true,
      rowStyles: `
        grid-template-columns:
        minmax(150px, 2fr)
        minmax(150px, 1fr);
      `,
    },
    {
      category: 'Research Institute Operations - 10%',
      description: '',
      totalCostUsd: '$12,250',
      removeEmptyCells: true,
      rowStyles: `
        grid-template-columns:
        minmax(150px, 2fr)
        minmax(150px, 1fr);
      `,
    },
    {
      category: 'Total',
      description: '',
      totalCostUsd: '$171,300',
      removeEmptyCells: true,
      rowStyles: `
        background: #F6F3FF;
        td { color: #8256FF; }
        grid-template-columns:
        minmax(150px, 2fr)
        minmax(150px, 1fr);
      `,
    },
  ],
  showHeaderRow: true,
  topPadding: true,
}

const Template: Story = (args) => <TableV2 {...defaultProps} {...args} />

export const Default = Template.bind({})
Default.args = {
  ...defaultProps,
  tableBodyStyles: `
    tr td:first-child,
    tr td: last-child {
      font-weight: 600;
    }
  `,
}

const Circle = styled.div`
  width: 17px;
  height: 17px;
  border-radius: 50%;
`

export const Legend = Template.bind({})
Legend.args = {
  columns: [
    {
      accessor: 'code',
      Header: 'Code',
    },
    { accessor: 'name', Header: 'Name' },
    { accessor: 'price', Header: 'price' },
  ],
  data: [
    {
      code: <Circle style={{ background: '#B8E8EB' }} />,
      name: 'Personnel',
      price: '$20,000',
    },
    {
      code: <Circle style={{ background: '#00C0CC' }} />,
      name: 'Equipment',
      price: '$20,000',
    },
    {
      code: <Circle style={{ background: '#004549' }} />,
      name: 'Materials and supplies',
      price: '$20,000',
    },
  ],
  tableStyles: `
    border: none;
    
    tr {
      grid-template-columns:
      minmax(30px, auto)
      minmax(150px, 1fr)
      minmax(50px, 1fr);

      td {
        display: flex;
        flex-direction: column;
        justify-content: center;
        padding: 0.5rem 0.3rem;
      }
    }

    tr:not(:last-child),
    thead tr {
      border-bottom: none;
    }
  `,
  showHeaderRow: false,
  topPadding: false,
}

const TextIconContainer = styled.div`
  align-items: center;
  display: flex;
  > *:not(:last-child) {
    margin-right: 0.3rem;
  }
`
const StyledIcon = styled(TooltipIcon)`
  @media ${device.laptop} {
    width: 1rem;
    height: 1rem;
  }
`

export const ProjectSummary = Template.bind({})
ProjectSummary.args = {
  columns: [
    {
      accessor: 'icon',
      Header: 'Icon',
    },
    { accessor: 'name', Header: 'Name' },
    { accessor: 'description', Header: 'Description' },
  ],
  data: [
    {
      icon: <img src="/assets/svgs/project-length.svg" />,
      name: 'Project Length',
      description: '1 year',
    },
    {
      icon: <img src="/assets/svgs/project-goal.svg" />,
      name: 'Goal',
      description:
        'Identify how stress impacts a mother’s brain circuitry and how that brain function affects parental behaviors.',
    },
    {
      icon: <img src="/assets/svgs/project-research-type.svg" />,
      name: 'Research Type',
      description: (
        <TextIconContainer>
          <p>Exploratory</p>
          <Tooltip>
            <StyledIcon
              alt={IMAGE_PROPS_INFO_PURPLE.getAlt()}
              src={IMAGE_PROPS_INFO_PURPLE.getSrc()}
            />
            <TooltipText>
              This amount combines what Aaron has donated + donations from his
              network
            </TooltipText>
          </Tooltip>
        </TextIconContainer>
      ),
    },
  ],
  tableStyles: `
    border: none;
    
    tr {
      grid-template-columns:
      minmax(30px, auto)
      minmax(100px, 1fr)
      minmax(100px, 1fr);

      td {
        padding: 0.5rem 0.3rem;

        &:first-child {
          img {
            height: 14px;
            width: 14px;
          }
        }
      }
    }

    @media ${device.tablet} {
      td {
        &:first-child img {
          margin-top: 0.3rem;
          height: 17px;
          width: 17px;
        }
      }
    }
  `,
  showHeaderRow: false,
  topPadding: false,
}
