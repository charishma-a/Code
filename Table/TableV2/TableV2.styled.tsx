import { HTMLAttributes } from 'react'
import styled, { css } from 'styled-components'
import { device } from '@/utils/device'

export const baseClass = 'kernls-table-v2'

export const Table = styled.table<
  HTMLAttributes<HTMLTableElement> & { styles?: string; topMargin?: boolean }
>`
  background-color: #fff;
  border: 1px solid #e5e5e5;
  border-radius: 4px;
  min-width: 100%;
  color: #565656;
  margin-top: ${(props) => (props.topMargin ? '1rem' : 0)};

  th {
    padding: 0.75rem 1rem;
  }
  td {
    padding: 0.75rem 1rem;
    /* overflow: hidden; */
    /* text-overflow: ellipsis; */
    /* white-space: nowrap; */

    p:first-child {
      margin-bottom: 0;
      margin-top: 0;
    }
  }

  th {
    /* position: sticky; */
    /* top: 0; */
    /* background: #6c7ae0; */
    text-align: left;
    font-weight: 600;
    font-size: 0.75rem;
  }

  th:last-child {
    border: 0;
  }

  td {
    padding-top: 10px;
    padding-bottom: 10px;
    font-size: 0.75rem;
    text-align: left;

    strong {
      color: #202020;
      font-weight: 600;
    }
  }

  tr:not(:last-child),
  thead tr {
    border-bottom: 1px solid #e5e5e5;
  }

  @media ${device.tablet} {
    th {
      font-size: 1rem;
    }
    td {
      font-size: 1rem;
    }
  }

  ${({ styles }) =>
    styles &&
    css`
      ${styles}
    `}
`

export const TableBody = styled.tbody<
  HTMLAttributes<HTMLTableSectionElement> & { styles?: string }
>`
  ${({ styles }) =>
    styles &&
    css`
      ${styles}
    `}
`

export const TableHeadCell = styled.th<
  HTMLAttributes<HTMLTableCellElement> & { styles?: string }
>`
  ${({ styles }) =>
    styles &&
    css`
      ${styles}
    `}
`

export const TableCell = styled.td<
  HTMLAttributes<HTMLTableCellElement> & { styles?: string }
>`
  ${({ styles }) =>
    styles &&
    css`
      ${styles}
    `}
`

export const TableHead = styled.thead<
  HTMLAttributes<HTMLTableSectionElement> & { styles?: string }
>`
  ${({ styles }) =>
    styles &&
    css`
      ${styles}
    `}
`

export const TableRow = styled.tr<
  HTMLAttributes<HTMLTableRowElement> & { isEmpty?: boolean; styles?: string }
>`
  display: grid;
  grid-template-columns: ${({ isEmpty }) => {
    if (isEmpty) {
      return `minmax(1fr, 1fr)`
    }
    return `minmax(90px, 1fr) minmax(90px, 1fr) minmax(90px, 1fr)`
  }};

  ${({ styles }) => styles || ''}
`

export const TableHeadRow = styled(TableRow)``
