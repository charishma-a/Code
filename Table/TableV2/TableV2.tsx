import React, { useMemo } from 'react'
import { Column, useTable } from 'react-table'

import * as sc from './TableV2.styled'

import { BaseComponentProps } from '@/common/types'

export interface CustomBaseColumn {
  removeEmptyCells?: boolean
  rowStyles?: string
}

export interface TableV2Props extends BaseComponentProps {
  columns: Array<Column<Record<string, unknown>>>
  data: Record<string, unknown>[]
  renderEmptyView?: () => React.ReactNode
  showHeaderRow?: boolean
  topPadding?: boolean
  tableBodyStyles?: string
  tableHeadStyles?: string
  tableStyles?: string
}

export const TableV2: React.FC<TableV2Props> = ({
  classNames,
  columns,
  data,
  renderEmptyView = () => null,
  showHeaderRow = true,
  topPadding,
  tableBodyStyles,
  tableHeadStyles,
  tableStyles,
  ...restProps
}) => {
  const className = [sc.baseClass, classNames].join(' ')

  const columnsMemo = useMemo(() => columns, [columns])
  const dataMemo = useMemo(() => data, [data])
  // Use the state and functions returned from useTable to build your UI
  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
    useTable({
      columns: columnsMemo,
      data: dataMemo,
    })

  // Render the UI for your table
  return (
    <sc.Table
      className={className}
      {...getTableProps()}
      styles={tableStyles}
      topMargin={topPadding}
      {...restProps}
    >
      {showHeaderRow ? (
        <sc.TableHead
          className={`${sc.baseClass}__thead`}
          styles={tableHeadStyles}
        >
          {headerGroups.map((headerGroup) => {
            return (
              <sc.TableHeadRow
                className={`${sc.baseClass}__row ${sc.baseClass}__thead__row`}
                key={headerGroup.id}
                {...headerGroup.getHeaderGroupProps()}
              >
                {headerGroup.headers.map((column) => {
                  return (
                    <sc.TableHeadCell
                      className={`${sc.baseClass}__cell ${sc.baseClass}__thead__row__cell`}
                      key={column.id}
                      {...column.getHeaderProps()}
                    >
                      {column.render('Header')}
                    </sc.TableHeadCell>
                  )
                })}
              </sc.TableHeadRow>
            )
          })}
        </sc.TableHead>
      ) : null}
      <sc.TableBody
        className={`${sc.baseClass}__tbody`}
        {...getTableBodyProps()}
        styles={tableBodyStyles}
      >
        {rows.length === 0 ? (
          <sc.TableRow
            className={`${sc.baseClass}__row ${sc.baseClass}__tbody__row--empty`}
            isEmpty
          >
            <sc.TableCell
              className={`${sc.baseClass}__cell ${sc.baseClass}__tbody__row__cell`}
            >
              {renderEmptyView()}
            </sc.TableCell>
          </sc.TableRow>
        ) : (
          rows.map((row) => {
            prepareRow(row)

            const { original } = row
            const { removeEmptyCells, rowStyles } = original as CustomBaseColumn

            return (
              <sc.TableRow
                className={`${sc.baseClass}__row ${sc.baseClass}__tbody__row`}
                key={row.id}
                styles={rowStyles}
                {...row.getRowProps()}
              >
                {row.cells.map((cell, index) => {
                  // const column: ColumnInstance<D> & CustomColumn = cell.column
                  if (removeEmptyCells && !cell.value) {
                    return null
                  }
                  return (
                    <sc.TableCell
                      className={`${sc.baseClass}__cell ${sc.baseClass}__tbody__row__cell`}
                      key={index}
                      {...cell.getCellProps()}
                    >
                      {cell.render('Cell')}
                    </sc.TableCell>
                  )
                })}
              </sc.TableRow>
            )
          })
        )}
      </sc.TableBody>
    </sc.Table>
  )
}
