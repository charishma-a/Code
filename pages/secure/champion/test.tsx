import { GetServerSideProps } from 'next'
import Link from 'next/link'
import React from 'react'
import { useSetAccessToken } from '@/hooks/useSetAccessToken'
import { LayoutAuth } from '@/components/LayoutAuth'
import { AuthPageCard } from '@/common/styled'
import { Paragraph } from '@/components/Paragraph'
import { fetchCommonSecureProps } from '@/utils/fetchCommonSecureProps'
import { BaseSecurePageProps } from '@/common/types'

const SecureChampionTest: React.FC<BaseSecurePageProps> = ({
  accessToken,
  redirectUrl,
}) => {
  useSetAccessToken({ accessToken, redirectUrl })
  if (!accessToken) {
    return null
  }
  return (
    <LayoutAuth
      description=""
      image=""
      title="Champion Dashboard | Kernls"
      twitterImage=""
      url="https://kernls.com/secure/champion/test"
    >
      <AuthPageCard>
        <Paragraph>This is a test page</Paragraph>
        <Link href="/secure/champion/dashboard">
          <a>dashboard</a>
        </Link>
      </AuthPageCard>
    </LayoutAuth>
  )
}

export const getServerSideProps: GetServerSideProps = async (context) => {
  const { accessToken, fetchResult, redirectUrl } =
    await fetchCommonSecureProps({
      context,
    })
  return { props: { ...fetchResult, accessToken, redirectUrl } }
}

export default SecureChampionTest
