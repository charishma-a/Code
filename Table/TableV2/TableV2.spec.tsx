import React from 'react'

import { TableV2 } from './TableV2'

import { render } from '@/test/testUtils'

const defaultProps = {
  columns: [
    {
      accessor: 'date',
      Header: 'Date',
      styles: `flex: 1 1 0; min-width: 20%;`,
    },
    { accessor: 'description', Header: 'Description', styles: `flex: 4 4 0;` },
  ],
  data: [
    {
      date: 'July 2021',
      description: 'Prepare subjects for neural recordings',
    },
    {
      date: 'Aug - Sept 2021',
      description: 'Collect neural recording data',
    },
    {
      date: 'Oct 2021',
      description: 'Analyze statistics',
    },
  ],
  showHeaderRow: true,
  topPadding: true,
}

describe('<TableV2 />', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<TableV2 {...defaultProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
