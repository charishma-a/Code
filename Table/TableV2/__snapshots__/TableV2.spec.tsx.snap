// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableV2 /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TableV2styled__Wrapper-gsywmi-8 dMsslf kernls-table-v2 "
  >
    <table
      class="TableV2styled__Table-gsywmi-0 fOSAye"
      role="table"
    >
      <thead
        class="TableV2styled__TableHead-gsywmi-4"
      >
        <tr
          class="TableV2styled__TableRow-gsywmi-5 TableV2styled__TableHeadRow-gsywmi-6 iBkcSj bGkSwo"
          role="row"
        >
          <th
            class="TableV2styled__TableHeadCell-gsywmi-2 cRcUDc"
            colspan="1"
            role="columnheader"
          >
            Date
          </th>
          <th
            class="TableV2styled__TableHeadCell-gsywmi-2 kliURf"
            colspan="1"
            role="columnheader"
          >
            Description
          </th>
        </tr>
      </thead>
      <tbody
        class="TableV2styled__TableBody-gsywmi-1"
        role="rowgroup"
      >
        <tr
          class="TableV2styled__TableRow-gsywmi-5 iBkcSj"
          role="row"
        >
          <td
            class="TableV2styled__TableCell-gsywmi-3 ffYjLu"
            role="cell"
          >
            July 2021
          </td>
          <td
            class="TableV2styled__TableCell-gsywmi-3 jFjynR"
            role="cell"
          >
            Prepare subjects for neural recordings
          </td>
        </tr>
        <tr
          class="TableV2styled__TableRow-gsywmi-5 iBkcSj"
          role="row"
        >
          <td
            class="TableV2styled__TableCell-gsywmi-3 ffYjLu"
            role="cell"
          >
            Aug - Sept 2021
          </td>
          <td
            class="TableV2styled__TableCell-gsywmi-3 jFjynR"
            role="cell"
          >
            Collect neural recording data
          </td>
        </tr>
        <tr
          class="TableV2styled__TableRow-gsywmi-5 iBkcSj"
          role="row"
        >
          <td
            class="TableV2styled__TableCell-gsywmi-3 ffYjLu"
            role="cell"
          >
            Oct 2021
          </td>
          <td
            class="TableV2styled__TableCell-gsywmi-3 jFjynR"
            role="cell"
          >
            Analyze statistics
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
